site/src/routes/pins-list.js

```diff
@@ -1,40 +1,199 @@
 import { HTTPError } from '../errors.js'
 import { verifyToken } from '../utils/utils.js'
-import * as PinataPSA from '../pinata-psa.js'
+import * as cluster from '../cluster.js'
 import { JSONResponse } from '../utils/json-response.js'
 
+/**
+ * @typedef {{
+ *   name?: string,
+ *   meta?: Record<string, string>,
+ *   pinStatus: import('../pinata-psa').Status,
+ *   size: number,
+ *   created: string
+ *   createdSort?: number
+ * }} NFTMetadata
+ * @typedef {{ name: string, metadata: NFTMetadata }} Key
+ */
+
 /**
  * @param {FetchEvent} event
  * @returns {Promise<JSONResponse>}
  */
-export async function pinsList (event) {
+export async function pinsList(event) {
   const result = await verifyToken(event)
   if (!result.ok) {
     return HTTPError.respond(result.error)
   }
   const { user } = result
   const { searchParams } = new URL(event.request.url)
 
-  const options = Object.fromEntries(searchParams.entries())
-  if (options.meta) {
-    let meta
-    try {
-      meta = JSON.parse(options.meta)
-      if (meta == null || typeof meta !== 'object') {
-        throw new Error('invalid meta')
+  let count = 0
+  /** @type {Key[]} */
+  let keys = []
+  const options = queryToListOptions(searchParams)
+  const limit = options.limit || 10
+  const filter = makeFilter(options)
+
+  let done = false
+  let cursor
+  while (!done) {
+    // @ts-ignore
+    const nftList = await stores.nfts.list({
+      prefix: user.sub,
+      cursor,
+      limit: 1000,
+    })
+    for (const key of nftList.keys) {
+      if (!filter(key)) continue
+      count++
+      key.metadata.createdSort = key.metadata.created
+        ? new Date(key.metadata.created).getTime()
+        : 0
```

Comment:
maybe we should put the `created` in seconds when we add nfts instead of iso string, we could run a migration to change it to `s` or `ms` plus make sure all of them have it.

Reply:
Yeah, this needs to be made good but we need a better database to do this right...